src/link/link_info/iptunnel.rs

```diff
@@ -202,13 +207,26 @@ impl<'a, T: AsRef<[u8]> + ?Sized> Parseable<NlaBuffer<&'a T>> for InfoIpTunnel {
                 parse_u32(payload)
                     .context("invalid IFLA_IPTUN_FLOWINFO value")?,
             ),
-            IFLA_IPTUN_FLAGS => Flags(TunnelFlags::from_bits_retain(
-                parse_u32(payload).context("invalid IFLA_IPTUN_FLAGS value")?,
-            )),
-            IFLA_IPTUN_PROTO => Protocol(IpProtocol::from(
-                parse_u8(payload).context("invalid IFLA_IPTUN_PROTO value")?
-                    as i32,
-            )),
+            IFLA_IPTUN_FLAGS => match kind {
+                super::InfoKind::SitTun => {
+                    // Parse as u16 for SIT tunnels
+                    Flags(TunnelFlags::from_bits_retain(u32::from(
+                        parse_u16(payload).context(
+                            "invalid IFLA_IPTUN_FLAGS value for SIT",
+                        )?,
+                    )))
+                }
+                _ => {
```

Comment:
There is no default in kernel for parsing this:

```
net/ipv4/ip_tunnel_core.c
1158:	if (data[IFLA_IPTUN_FLAGS]) {
1161:		flags = nla_get_be16(data[IFLA_IPTUN_FLAGS]);

net/ipv6/ip6_tunnel.c
1992:	if (data[IFLA_IPTUN_FLAGS])
1993:		parms->flags = nla_get_u32(data[IFLA_IPTUN_FLAGS]);
2104:		/* IFLA_IPTUN_FLAGS */
2134:	    nla_put_u32(skb, IFLA_IPTUN_FLAGS, parm->flags) ||
2170:	[IFLA_IPTUN_FLAGS]		= { .type = NLA_U32 },

net/ipv6/sit.c
1668:		/* IFLA_IPTUN_FLAGS */
1708:	    nla_put_be16(skb, IFLA_IPTUN_FLAGS,
1748:	[IFLA_IPTUN_FLAGS]		= { .type = NLA_U16 },
```

Always do explicit kind matching and fail for unknown kind.

Reply:
I added the explicit kind matching and fail for an unknown kind.

Reply:
Both SIT and IPIP are using `nla_put_be16()`, not native endia